### DISCUSSION TITLE
User name instead of phone number

### DISCUSSION THREAD

**Discussion by seyfeb** [bbernhard/signal-cli-rest-api#499] [Q&A] [answered]
In the documentation it is described that you need to set up a mobile phone number in order to use the API.

https://github.com/bbernhard/signal-cli-rest-api/blob/master/doc/HOMEASSISTANT.md#set-up-a-phone-number

Signal recently [introduced usernames](https://signal.org/blog/phone-number-privacy-usernames/) but I do not understand all implication this has. Do you think it will be possible to use the CLI without the need of setting up a Signal account with a phone number but using a user name instead?

**Comment by bbernhard** [comment, marked as answer]
As far as I know, a phone number is still required for registration (this is also mentioned in the linked article). But it is now possible to hide the phone number and connect with other people via the username.

**Reply by seyfeb** [reply]
You're right, I missed the part where it says that the number is still required for signup!

**Reply by KarelTa** [reply]
from version Signal 8.28.2 (beta now), will be possible create an account without phone no. It will be a great solution. will be possible to install as "full" app - add on and create the account in HA directly?

**Comment by mattdale77** [comment]
A related question about how this impacts being able to send messages to people when they haven't shared a number. Is it possible to just replace the recipient number with a username in v2/send ?
I've seen that it's possible to send to a UUID in Signal API, is there a way to look this up for a user in contacts so if they decide to hide their number we will still have a handle to send messages with?

**Reply by Qhilm** [reply]
yep, I tested, it works, you can send to the username.
